src/modules/kvs/kvs.c

```diff
@@ -73,6 +74,11 @@ struct kvs_ctx {
     bool events_init;            /* flag */
     const char *hash_name;
     unsigned int seq;           /* for commit transactions */
+    double sync;                /* in seconds */
+    flux_watcher_t *sync_w;
+    bool sync_timer_started;
+    bool sync_txn_submitted;
```

Comment:
ahh good catch, i don't allow two "sync prepends" to the kvstxn queue, so it literally would do nothing if a second were attempted to be added by the user.

Reply:
oh wait ... this was there to catch a corner case.  a "sync prepend" may occur while a commit is in progress, so the checkoutpoint would occur right after the current commit in progress is complete.

If we removed this flag, the sync timer could go off for the commit right before the checkpoint is done.  I guess we could allow it?   But it seems sort of silly to me to allow this.  We'd want the sync timer to go off for the commit right after the sync/checkpoint.

Reply:
I'm having a hard time following this, and that's not a good indication that someone coming along later will be able to quickly understand it enough to make changes.  I'm trying to think what would cordon this feature off from the rest of the complex kvs stuff, so it could be grokked in isolation.

What if we just create a new source module sync.c and each time we _complete_ a transaction on the root namespace, we call a function like `sync_setroot()`.  Then in that module (without hooking into any other part of the KVS except maybe a configuration callback) we implement the content flush + checkpoint and timer?  Maybe sync.c could contain functions that are also used by the FLUX_KVS_SYNC flag handling in the commit code?

Too much of a do-over?

Reply:
another thought, would it be worthwhile to go back to the earlier timer based method?  Simply wake up every X time and perform a sync using FLUX_KVS_SYNC?  So much of the complexity is eliminated, but at the cost of an occasional timer wakeup and the setroot event that goes out every once in awhile.

Reply:
It  seems wrong to broadcast the setroot event when  the KVS content has not changed.

Doing this standalone really shouldn't be that bad - in fact, couldn't it just subscribe to the setroot event and use that to start the timer?